File: src/commands/ping.ts
import { invoke } from "@tauri-apps/api/core";
import type { AppResponse } from "./types";

export interface PingBackendData {
  message: string;
}

export function pingBackend(): Promise<AppResponse<PingBackendData>> {
  return invoke<AppResponse<PingBackendData>>("ping_backend");
}

File: src-tauri/src/commands/ping.rs
use serde::Serialize;

use crate::types::response::AppResponse;

#[derive(Debug, Serialize)]
pub struct PingData {
    message: String,
}

#[tauri::command]
pub fn ping_backend() -> AppResponse<PingData> {
    AppResponse::ok(PingData {
        message: "pong".to_string(),
    })
}